ncm-grub/src/main/perl/grub.pm

```diff
@@ -191,11 +193,25 @@ sub password
     my $val = $tree->{option} ? "--$tree->{option} " : "";
     $val .= $password;
 
-    $grub_fh->add_or_replace_lines(qr/^password\s+/,
-                                   qr/^password $val$/,
-                                   "password $val\n",
-                                   $self->main_section_offset($grub_fh),
-                                   );
+    if ( -d $GRUB2_DIR ) {
+        my $usercfg_fh = CAF::FileEditor->new( $GRUB2_USER_CFG,
+            owner => "root",
+            group => "root",
+            mode  => 0400,
+            log   => $self );
+        
+        $usercfg_fh->add_or_replace_lines(qr/^GRUB_PASSWORD=/, qr/^GRUB_PASSWORD=$password$/, "GRUB_PASSWORD=$password\n");
+        $usercfg_fh->close();
```

Comment:
this is not suficient on it's own i think. doesn't this need a `grub-update`?
btw, for serial console support, you can modifiy `/etc/default/grub` (https://wiki.archlinux.org/index.php/working_with_the_serial_console) and do an update.